db/migrate/20200428205535_uniqueness_rule_references_rule_index.rb

```diff
@@ -0,0 +1,28 @@
+class UniquenessRuleReferencesRuleIndex < ActiveRecord::Migration[5.2]
+  class NewRuleReferencesRule < ApplicationRecord; end
+
+  def change
+    ids_to_keep = RuleReferencesRule.group(
+      'rule_id, rule_reference_id'
+    ).having("count(*) > 1").pluck(Arel.sql('MIN(id)'))
+    create_table :new_rule_references_rules, id: false do |t|
+      t.references :rule, type: :uuid, index: true, null: false
+      t.references :rule_reference, type: :uuid, index: true, null: false
+    end
+    columns = [:rule_id, :rule_reference_id]
+    batch_size = 1000
+    to_keep = RuleReferencesRule.where(id: ids_to_keep)
+    batches_to_run = to_keep.count/batch_size
+    to_keep.in_batches(of: batch_size).each_with_index do |batch, index|
+      puts "Inserting RRR batch, #{index} out of #{batches_to_run}"
+      NewRuleReferencesRule.import(
+        columns,
+        batch.pluck(:rule_id, :rule_reference_id)
```

Comment:
I'd be curious about select vs. pluck here as well.

Reply:
AR relations can't be passed to the `.insert` method, it throws an InvalidArgumentsError.

Reply:
Geez, maybe it'd make more sense to write an `INSERT` query yourself with an AR subquery using `to_sql`

Reply:
I really doubt trying to do this like I suggested would make any difference. We'd still need to `pluck` to get all the records to insert.